ch4/exercise4-2/printHash.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"bufio"
+	"crypto/sha256"
+	"crypto/sha512"
+	"flag"
+	"fmt"
+	"os"
+)
+
+func main() {
+
+	var s384 = flag.Bool("sha384", false, "generate sha384 hash")
+	var s512 = flag.Bool("sha512", false, "generate sha512 hash")
+	flag.Parse()
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Input: ")
+	input, _ := reader.ReadString('\n')
```

Comment:
For practicality/integration with other cli utilities, I'd recommend doing something like you did earlier:

https://github.com/quinqu/tgpl/blob/2b0357e865bae9430a6ba0b72d166c1018a2669d/ch1/exercise4.go#L14-L18

This is a more UNIX-y approach to IO than prompting for a newline delimited string.  For instance, what happens with your program when I want to hash some data (e.g. a file) that has a newline in it?